server/queues/processors/NotificationsProcessor.ts

```diff
@@ -182,3 +225,85 @@ export default class NotificationsProcessor extends BaseProcessor {
     }
   }
 }
+
+type Notifiable = {
+  user: User;
+  userId: string;
+  document: Document;
+  documentId: string;
+  event: DocumentActionEvent | RevisionEvent;
+  eventName: string;
+};
+
+const shouldNotify = async (subject: Notifiable): Promise<boolean> => {
```

Comment:
Lets make this a private method on the class rather than a function floating in the file